### ISSUE TITLE
Allow "no-restricted-properties" to check for in-depth properties

### ISSUE THREAD

**Issue by eTorAken** [eslint/eslint#8505]
**What rule do you want to change?**
no-restricted-properties

**Does this change cause the rule to produce more or fewer warnings?**
More

**How will the change be implemented? (New option, new default behavior, etc.)?**
No change but `property` can handle an in-depth properties chain.

**Please provide some example code that this change will affect:**
I wish to use this with expect statements to avoid bad english practices:
```js
/* eslint no-restricted-properties: [2, {
    "property": "to.be.equal",
    "message": "Equal is a verb, use 'to.equal' instead",
}] */

expect(this).to.be.equal(that); // error
expect(this).to.equal(that); // ok
expect(this).to.deep.equal(that); // ok
```

**What does the rule currently do for this code?**
Does not throw when "be" and "equal" are used one before another.

**What will the rule do after it's changed?**
Throw an error when "equal" property follows "be" property.

**Comment by darkyndy** [comment]
I did this using `no-restricted-syntax`.

```
"no-restricted-syntax": [2,
      {
        // test for .to.be.equal
        "selector": "MemberExpression[object.type=MemberExpression][property.type=Identifier][property.name=equal] > MemberExpression[object.type=MemberExpression][property.type=Identifier][property.name=be] > MemberExpression[property.type=Identifier][property.name=to]",
        "message": "Equal is a verb, use 'to.equal' instead"
      }
    ],
```

other selector that I made for this rule is:
```
"selector": "MemberExpression[object.type=MemberExpression][object.object.type=MemberExpression][object.object.property.type=Identifier][object.object.property.name=to][object.property.type=Identifier][object.property.name=be][property.type=Identifier][property.name=equal]",
// I do not like this one
```

I know that doesn't look nice (needs improvement), but it works.

I like your proposal, will make things a lot easier.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
